rembg/sessions/sam.py

```diff
@@ -105,9 +105,10 @@ def __init__(
         valid_providers = []
         available_providers = ort.get_available_providers()
 
-        for provider in providers or []:
-            if provider in available_providers:
-                valid_providers.append(provider)
+        if providers:
+            for provider in providers or []:
+                if provider in available_providers:
+                    valid_providers.append(provider)
```

Comment:
Added `if providers:` like other models.
This was causing unintended fallbacks.

Reply:
This appears to have changed with this commit: https://github.com/danielgatis/rembg/commit/6f1dd31ee68f3e7613563bdeef86a2b319896ccc